specs/signature-specification.md

```diff
@@ -258,17 +258,22 @@ The signing certificate's public key algorithm and size MUST be used to determin
 
 ### Certificate Requirements
 
-The codesigning and timestamping certificates MUST meet the following requirements. These requirements are validated both at signature generation time and signature verification time, and are applied to the certificate chain in the signature envelope. These validations are independent of certificate chain validation against a trust store.
+The **codesigning** certificates MUST meet the following requirements. These requirements are validated both at signature generation time and signature verification time, and are applied to the certificate chain in the signature envelope. These validations are independent of certificate chain validation against a trust store.
+
+The **timestamping** certificates MUST meet the following requirements as well. Requirements for leaf certificate are validated both at timestamp token generation and verification time. Requirements for root and intermediate CA certificates are only validated at timestamp token verification time. This is because at generation time, the client is NOT required to have access to the full certificate chain of the Timestamp Authority (TSA). These validations are independent of certificate chain validation against a trust store.
 
 #### Root and Intermediate CA Certificates
 
 The CA certificates MUST meet the following requirements
 
 1. **[Basic Constraints:](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.9)**
-The `basicConstraints` extension MUST be present and MUST be marked as critical. The `cA` field MUST be set `true`.
+The `basicConstraints` extension MUST be present and MUST be marked as critical. The `cA` field MUST be set to `true`.
 The [`pathLenConstraint`](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.9) field is OPTIONAL. If present, it MUST be verified against the depth of the chain below that CA certificate. (If value is null consider it as not present)
 1. **[Key Usage:](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.3)**
-The `keyUsage` extension MUST be present and MUST be marked critical. Bit positions for `keyCertSign` MUST be set.
+    - **For codesigning certificate:** the `keyUsage` extension MUST be present and MUST be marked critical.
+    - **For timestamping certificate:** the `keyUsage` extension MUST be present and SHOULD be marked critical.
```

Comment:
@priteshbandi We found some TSAs are not marking this as critical. Because, for timestamping cert, there's the `extended key usage field extension` with KeyPurposeID value ` id-kp-timeStamping`, this extension MUST be marked as critical according to RFC 3161: https://datatracker.ietf.org/doc/html/rfc3161#section-2.3.
To allow Notary Project able to use such TSAs mentioned above, we can only enforce the check of `id-kp-timeStamping` as RFC compliant.

Reply:
@priteshbandi This is for compatibility reasons. We did some tests with existing TSAs and found some TSAs do not mark `keyUsage` as critical.

Reply:
for posterity, can you please share, list of TSA tested and there results?

Reply:
> for posterity, can you please share, list of TSA tested and there results?


Hi @priteshbandi, @Two-Hearts will provide a list of tested TSAs combining results of certificate chain testing based on our community discussion on Jun 10 2024. For the existing TSAs that `keyUsage` of root CA certificate is not marked critical, see the root CA certificate named "Microsoft Root Certificate Authority 2010" listed in this page https://www.microsoft.com/pkiops/docs/repository.htm. Hope it helps.